### PR TITLE
doc: test_and_optimize: debugging tools list edits

### PR DESCRIPTION
**Pull request by greg-fer**
Reworked the debugging tools section into a table with more tools listed. NCSDK-22325.

----

- [x] Changelog entry before merge.
- [x] Rebase before merge.

### CODE REVIEW

**Comment by peknis** [comment]
Changes to links.txt implemented in https://github.com/nrfconnect/sdk-nrf/pull/14650.

**Review by peknis — APPROVED** [review]

**Comment by greg-fer** [comment]
> Changes to links.txt implemented in #14650.

Thanks! I'll rebase before requesting merge.

**Review by hellesvik-nordic — APPROVED** [review]
I would mention PPK2. Measuring current is often a good way to debug stuff.
Other than that, this is great; I love lists!

**Comment by hellesvik-nordic** [comment]
> I would mention PPK2. Measuring current is often a good way to debug stuff. Other than that, this is great; I love lists!

I see now that the Power Profiler app is mentioned, which covers the PPK2. 
So this is simply just great, I got nothing to add

File: doc/nrf/test_and_optimize/debugging.rst [hunk 1]
```diff
@@ -133,10 +133,42 @@ Use the following steps to enable monitor-mode debugging in the |NCS|:
...
+     - :ref:`ug_bt`
+   * - `Cellular Monitor`_
+     - Capture and analyze modem traces to evaluate communication and view network parameters. Available from `nRF Connect for Desktop`_.
+     - :ref:`ug_lte`
+   * - `nRF Connect Direct Test Mode`_
+     - Perform RF PHY checks of Bluetooth Low Energy devices using a GUI for the Bluetooth-specified Direct Test Mode. Available from `nRF Connect for Desktop`_.
+     - :ref:`ug_bt`
+   * - `nRF Connect Power Profiler`_
+     - Measure the real-time power consumption of your designs. Available from `nRF Connect for Desktop`_.
+     - :ref:`ug_bt`, :ref:`ug_lte`, :ref:`ug_matter`, :ref:`ug_thread`, :ref:`ug_wifi`
```
Comment (edmont) [inline doc/nrf/test_and_optimize/debugging.rst:162]:
```suggestion
     - :ref:`ug_bt`, :ref:`ug_lte`, :ref:`ug_matter`, :ref:`ug_thread`, :ref:`ug_wifi`, :ref:`ug_zigbee`
```

**Review by edmont — APPROVED** [review]

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request merged**